src/app/ipv/router.js

```diff
@@ -49,6 +51,17 @@ router.post(
   handleJourneyAction,
 );
 
+// Special case to handle determination of COI journey type based on the checkboxes selected and determine the error type
+router.post(
+  getPagePath(CONFIRM_DETAILS),
+  parseForm,
+  csrfProtection,
+  formHandleUpdateDetailsCheckBox,
```

Comment:
thanks, yes I thought about that but would have to add a check in `formHandleUpdateDetailsCheckBox` to check if the journey != undefined (in the case of the user selects that their details are already correct) so it doesn't get overwritten, so wasn't sure what the best approach was

Reply:
I see the dilemma. Could it make sense to use `formHandleUpdateDetailsCheckBox` inside `formHandleCoiDetailsCheck`? Since we only need to use the checkbox handler in the case where `(req.body.detailsCorrect === "no" && req.body.detailsToUpdate)`

Reply:
good shout, will add